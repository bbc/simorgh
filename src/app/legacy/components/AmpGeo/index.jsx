import React from 'react';

const configuration = {
  AmpBind: true,
  ISOCountryGroups: {
    eea: [
      'at',
      'ax',
      'be',
      'bg',
      'bl',
      'cy',
      'cz',
      'de',
      'dk',
      'ea',
      'ee',
      'es',
      'fi',
      'fr',
      'gb',
      'gf',
      'gg',
      'gi',
      'gp',
      'gr',
      'hr',
      'hu',
      'ic',
      'ie',
      'im',
      'is',
      'it',
      'je',
      'li',
      'lt',
      'lu',
      'lv',
      'mf',
      'mq',
      'mt',
      'nc',
      'nl',
      'no',
      'pf',
      'pl',
      'pm',
      'pt',
      're',
      'ro',
      'se',
      'si',
      'sj',
      'sk',
      'tf',
      'va',
      'wf',
      'yt',
    ],
    gbOrUnknown: ['gb', 'gg', 'im', 'je', 'uk', 'unknown'],
  },
};

export const AMP_GEO_SCRIPT = (
  <script
    async
    custom-element="amp-geo"
    src="https://cdn.ampproject.org/v0/amp-geo-0.1.js"
  />
);

const AmpGeo = () => (
  <amp-geo layout="nodisplay">
    <script
      type="application/json"
      // eslint-disable-next-line react/no-danger
      dangerouslySetInnerHTML={{ __html: JSON.stringify(configuration) }}
    />
  </amp-geo>
);

export default AmpGeo;
