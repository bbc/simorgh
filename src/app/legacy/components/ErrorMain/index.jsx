import React from 'react';
import { string, arrayOf, shape } from 'prop-types';
import styled from '@emotion/styled';
import InlineLink from '#psammead/psammead-inline-link/src';
import Paragraph from '#psammead/psammead-paragraph/src';
import { getSerifMedium } from '#psammead/psammead-styles/src/font-styles';
import {
  getCanon,
  getParagon,
  GEL_FF_REITH_SANS,
} from '#psammead/gel-foundations/src/typography';
import idSanitiser from '#lib/utilities/idSanitiser';
import Grid, { GelPageGrid } from '#components/Grid';
import { scriptPropType } from '#psammead/gel-foundations/src/prop-types';

const StatusCode = styled.span`
  ${props => (props.script ? getParagon(props.script) : '')}
  color: ${props => props.theme.palette.POSTBOX};
  display: block;
  font-family: ${GEL_FF_REITH_SANS};
  font-weight: 600;
  padding: 2.5rem 0 0.5rem 0;
`;

const Heading = styled.h1`
  ${({ script }) => script && getCanon(script)}
  ${({ service }) => getSerifMedium(service)}
  color: ${props => props.theme.palette.SHADOW};
  margin-top: 0;
`;

const StyledGelPageGrid = styled(GelPageGrid)`
  padding-bottom: 4rem;
`;

const CustomParagraph = styled(Paragraph)`
  padding-top: 0.2rem;
`;

const ErrorMain = ({
  statusCode,
  title,
  message,
  solutions,
  callToActionFirst,
  callToActionLinkText,
  callToActionLinkUrl,
  callToActionLast,
  script,
  service,
}) => (
  <StyledGelPageGrid
    as="main"
    role="main"
    columns={{
      group0: 6,
      group1: 6,
      group2: 6,
      group3: 6,
      group4: 8,
      group5: 20,
    }}
    enableGelGutters
  >
    <Grid
      item
      startOffset={{
        group0: 1,
        group1: 1,
        group2: 1,
        group3: 1,
        group4: 2,
        group5: 5,
      }}
      columns={{
        group0: 6,
        group1: 6,
        group2: 6,
        group3: 6,
        group4: 6,
        group5: 12,
      }}
      margins={{ group0: true, group1: true, group2: true, group3: true }}
    >
      <StatusCode script={script} data-e2e="status-code">
        {statusCode}
      </StatusCode>
      <Heading id="content" script={script} service={service} tabIndex="-1">
        {title}
      </Heading>
      <CustomParagraph script={script} service={service}>
        {message}
      </CustomParagraph>
      <ul>
        {solutions.map(text => (
          <CustomParagraph
            script={script}
            service={service}
            as="li"
            key={idSanitiser(text)}
          >
            {text}
          </CustomParagraph>
        ))}
      </ul>
      <CustomParagraph script={script} service={service}>
        {callToActionFirst}
        <InlineLink
          href={callToActionLinkUrl}
          className="focusIndicatorReducedWidth"
        >
          {callToActionLinkText}
        </InlineLink>
        {callToActionLast}
      </CustomParagraph>
    </Grid>
  </StyledGelPageGrid>
);

ErrorMain.propTypes = {
  statusCode: string.isRequired,
  title: string.isRequired,
  message: string.isRequired,
  solutions: arrayOf(string).isRequired,
  callToActionFirst: string,
  callToActionLinkText: string.isRequired,
  callToActionLinkUrl: string.isRequired,
  callToActionLast: string,
  script: shape(scriptPropType).isRequired,
  service: string.isRequired,
};

ErrorMain.defaultProps = {
  callToActionFirst: null,
  callToActionLast: null,
};

export default ErrorMain;
