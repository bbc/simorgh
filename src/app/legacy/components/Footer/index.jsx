import React from 'react';
import styled from '@emotion/styled';
import { arrayOf, shape, string, node, bool } from 'prop-types';
import {
  getSansBold,
  getSansRegular,
} from '#psammead/psammead-styles/src/font-styles';
import { getBrevier } from '#psammead/gel-foundations/src/typography';
import {
  GEL_SPACING,
  GEL_SPACING_DBL,
  GEL_MARGIN_BELOW_400PX,
  GEL_MARGIN_ABOVE_400PX,
} from '#psammead/gel-foundations/src/spacings';
import {
  GEL_GROUP_2_SCREEN_WIDTH_MIN,
  GEL_GROUP_4_SCREEN_WIDTH_MIN,
} from '#psammead/gel-foundations/src/breakpoints';

import { AmpCookieSettingsButton } from '#containers/ConsentBanner/Banner/cookie.amp';
import Link from './Link';
import List from './List';

const SitewideLinksWrapper = styled.div`
  ${({ script }) => script && getBrevier(script)}
  ${({ service }) => service && getSansRegular(service)}
  background-color: ${props => props.theme.palette.EBON};

  @media (max-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    padding: 0 ${GEL_MARGIN_BELOW_400PX};
  }
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    padding: 0 ${GEL_MARGIN_ABOVE_400PX};
  }
`;

const ConstrainedWrapper = styled.div`
  max-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN};
  margin: 0 auto;
  ${({ trustProjectLink }) =>
    trustProjectLink && `padding-top: ${GEL_SPACING};`}
`;

const StyledParagraph = styled.p`
  color: ${props => props.theme.palette.WHITE};
  margin: 0;
  padding: ${GEL_SPACING_DBL} 0;

  // removes padding which creates touch target from the final "inline" link so the Focus Indicator doesn't obscure other text.
  a {
    padding: 0;
  }
`;

const StyledAmpCookieSettingsButton = styled(AmpCookieSettingsButton)`
  ${({ service }) => service && getSansBold(service)}
  background: none;
  border: none;
  color: ${props => props.theme.palette.WHITE};
  cursor: pointer;
  display: block;
  padding: ${GEL_SPACING} 0 ${GEL_SPACING};
  text-decoration: none;
  text-align: left;
  width: 100%;

  &:hover,
  &:focus {
    text-decoration: underline;
  }
`;

const SitewideLinks = ({
  links,
  trustProjectLink,
  copyrightText,
  externalLink,
  isAmp,
  script,
  service,
}) => {
  const elements = links.map(({ id, text, href, lang }) => {
    if (isAmp && id === 'COOKIE_SETTINGS') {
      return (
        <StyledAmpCookieSettingsButton lang={lang} service={service}>
          {text}
        </StyledAmpCookieSettingsButton>
      );
    }
    return <Link service={service} text={text} href={href} lang={lang} />;
  });

  return (
    <SitewideLinksWrapper script={script} service={service}>
      <ConstrainedWrapper trustProjectLink={trustProjectLink}>
        <List
          service={service}
          elements={elements}
          trustProjectLink={trustProjectLink}
        />
        <StyledParagraph>
          {copyrightText}{' '}
          <Link
            service={service}
            text={externalLink.text}
            href={externalLink.href}
            inline
          />
        </StyledParagraph>
      </ConstrainedWrapper>
    </SitewideLinksWrapper>
  );
};

const linkPropTypes = shape({
  href: string.isRequired,
  text: string.isRequired,
});

SitewideLinks.propTypes = {
  links: arrayOf(linkPropTypes.isRequired).isRequired,
  copyrightText: node.isRequired,
  trustProjectLink: linkPropTypes,
  externalLink: linkPropTypes.isRequired,
  isAmp: bool,
  script: shape({}),
  service: string,
};

SitewideLinks.defaultProps = {
  script: null,
  service: null,
  isAmp: false,
  trustProjectLink: null,
};

export default SitewideLinks;
