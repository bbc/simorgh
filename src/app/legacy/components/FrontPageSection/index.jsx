import styled from '@emotion/styled';

// Components
import { GEL_GROUP_4_SCREEN_WIDTH_MIN } from '#psammead/gel-foundations/src/breakpoints';

const FrontPageSection = styled.div`
  /* To centre page layout for Group 4+ */
  margin: 0 auto;
  width: 100%; /* Needed for IE11 */
  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
    max-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN};
  }
`;

export default FrontPageSection;
