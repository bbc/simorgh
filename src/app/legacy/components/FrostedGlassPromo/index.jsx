import React, { useContext } from 'react';
import { shape, node, string, number } from 'prop-types';
import styled from '@emotion/styled';
import pick from 'ramda/src/pick';
import Lazyload from 'react-lazyload';

import { RequestContext } from '#contexts/RequestContext';
import useClickTrackerHandler from '#hooks/useClickTrackerHandler';
import { getSerifRegular } from '#psammead/psammead-styles/src/font-styles';
import { GEL_GROUP_2_SCREEN_WIDTH_MIN } from '#psammead/gel-foundations/src/breakpoints';
import {
  GEL_SPACING,
  GEL_SPACING_DBL,
} from '#psammead/gel-foundations/src/spacings';
import makeRelativeUrlPath from '#lib/utilities/makeRelativeUrlPath';
import ImageWithPlaceholder from '#containers/ImageWithPlaceholder';
import { ServiceContext } from '../../../contexts/ServiceContext';

import FrostedGlassPanel from './FrostedGlassPanel';

import withData from './withData';

const PANEL_OFFSET = 250;

const Wrapper = styled.div`
  position: relative;
  width: 100%;
  display: flex;
  flex-direction: column;
  height: 100%;
  text-decoration: none;
  &:hover {
    a {
      text-decoration: underline;
    }
  }
  &:visited {
    a {
      color: #e6e8ea;
    }
  }
`;

// This is an unfocusable element, hidden from screenreaders
// This is to allow mouse users to click anywhere on the promo
// But for keyboard and AT users, and scenarios where there is no CSS,
// only the link inside the H3 should be interactive
const ClickableArea = styled.a`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 5;
`;

const H3 = styled.h3`
  margin: 0;
`;

const A = styled.a`
  display: inline-block;
  ${({ service }) => service && getSerifRegular(service)}
  text-decoration: none;
  color: ${props =>
    props.isAmp ? props.theme.palette.BLACK : props.theme.palette.WHITE};
  font-size: 0.9375rem;
  line-height: 1.33;
  margin: 0.875rem ${GEL_SPACING} 0 ${GEL_SPACING};
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    font-size: 1rem;
    line-height: 1.25;
    margin: 0.875rem ${GEL_SPACING_DBL} 0 ${GEL_SPACING_DBL};
  }
  &:visited {
    color: ${props =>
      props.isAmp ? props.theme.palette.METAL : props.theme.palette.GREY_3};
  }
  &:focus {
    text-decoration: underline;
  }
`;

const LazyloadPlaceholder = styled.div`
  background-color: ${props =>
    props.isAmp ? props.theme.palette.WHITE : props.theme.palette.GREY_8};
  min-height: 100px;
  padding-bottom: ${GEL_SPACING_DBL};
`;

const LazyloadWrapper = styled(Lazyload)`
  height: 100%;
`;

const FrostedGlassPromo = ({
  image,
  children,
  footer,
  url,
  eventTrackingData,
  index,
  minimumContrast,
  paletteSize,
}) => {
  const { script, service } = useContext(ServiceContext);
  const { isAmp } = useContext(RequestContext);
  const isCanonical = !isAmp;
  const relativeUrl = makeRelativeUrlPath(url);

  const clickTracker = useClickTrackerHandler({
    ...(eventTrackingData || {}),
    url: relativeUrl,
  });

  const onClick = eventTrackingData ? clickTracker : () => {};

  const promoText = (
    <>
      <H3>
        <A
          script={script}
          service={service}
          href={relativeUrl}
          onClick={onClick}
          isAmp={isAmp}
        >
          {children}
        </A>
      </H3>
      {footer}
    </>
  );

  // The ClickableArea component is an anchor ("a") element
  // Anchors cannot be self-closing under the HTML spec
  /* eslint-disable react/self-closing-comp */
  return (
    <Wrapper data-testid={`frosted-promo-${index}`}>
      <ClickableArea
        href={relativeUrl}
        onClick={onClick}
        aria-hidden="true"
        tabIndex="-1"
      ></ClickableArea>
      <ImageWithPlaceholder
        lazyLoad
        darkMode={isCanonical}
        {...pick(
          [
            'src',
            'srcset',
            'fallbackSrcset',
            'primaryMimeType',
            'fallbackMimeType',
            'sizes',
            'alt',
            'ratio',
            'width',
            'height',
          ],
          image,
        )}
      />
      <LazyloadWrapper
        offset={PANEL_OFFSET}
        once
        placeholder={
          // Placeholder always gets rendered on AMP
          <LazyloadPlaceholder
            data-testid="frosted-glass-lazyload-placeholder"
            isAmp={isAmp}
          >
            {promoText}
          </LazyloadPlaceholder>
        }
      >
        <FrostedGlassPanel
          image={image.src}
          minimumContrast={minimumContrast}
          paletteSize={paletteSize}
        >
          {promoText}
        </FrostedGlassPanel>
      </LazyloadWrapper>
    </Wrapper>
  );
};

// This component can receive promo data in a range of different formats
// It uses a withData HoC to convert the prop to a standardised schema
// This array is the list of props that should just be passed straight through
// to the component, without requiring any preprocessing
const propsToPassThrough = ['minimumContrast', 'paletteSize', 'isAmp'];

FrostedGlassPromo.propTypes = {
  children: node.isRequired,
  url: string.isRequired,
  footer: node,
  eventTrackingData: shape({}),
  index: number,
  minimumContrast: number,
  paletteSize: number,
  image: shape({
    src: string.isRequired,
    alt: string.isRequired,
    ratio: number.isRequired,
    width: number.isRequired,
    height: number.isRequired,
    smallSrc: string,
    srcset: string,
    sizes: string,
  }).isRequired,
};

FrostedGlassPromo.defaultProps = {
  footer: null,
  eventTrackingData: null,
  index: 0,
  minimumContrast: 8,
  paletteSize: 10,
};

export default withData(FrostedGlassPromo, propsToPassThrough);
