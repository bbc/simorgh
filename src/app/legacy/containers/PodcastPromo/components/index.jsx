import React from 'react';
import styled from '@emotion/styled';
import { string, shape, arrayOf, element } from 'prop-types';
import { scriptPropType } from '#psammead/gel-foundations/src/prop-types';
import { GEL_SPACING_DBL } from '#psammead/gel-foundations/src/spacings';

import Title from './title';
import Card from './card';
import CardContent from './card-content';
import CardLink from './card-link';
import CardImageWrapper from './card-image-wrapper';
import CardTitle from './card-title';
import CardDescription from './card-description';
import CardEpisodesText from './card-episodes-text';

const PodcastContext = React.createContext({});
const withPodcastContext = Component => props =>
  (
    <PodcastContext.Consumer>
      {context => <Component {...context} {...props} />}
    </PodcastContext.Consumer>
  );

const Wrapper = styled.section`
  background-color: ${props => props.theme.palette.LUNAR};
  padding: ${GEL_SPACING_DBL};
`;

const PodcastPromo = ({ script, service, children, ...props }) => (
  <PodcastContext.Provider value={{ script, service }}>
    <Wrapper {...props}>{children}</Wrapper>
  </PodcastContext.Provider>
);

PodcastPromo.Title = withPodcastContext(Title);
PodcastPromo.Card = Card;
PodcastPromo.Card.Link = CardLink;
PodcastPromo.Card.ImageWrapper = CardImageWrapper;
PodcastPromo.Card.Content = CardContent;
PodcastPromo.Card.Title = withPodcastContext(CardTitle);
PodcastPromo.Card.Description = withPodcastContext(CardDescription);
PodcastPromo.Card.EpisodesText = withPodcastContext(CardEpisodesText);

PodcastPromo.propTypes = {
  children: arrayOf(element).isRequired,
  script: shape(scriptPropType).isRequired,
  service: string.isRequired,
};

PodcastPromo.defaultProps = {};

export default PodcastPromo;
