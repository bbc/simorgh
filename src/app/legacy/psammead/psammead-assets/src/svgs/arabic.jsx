import React from 'react';

const height = 24.59;
const width = 222.3514;
const arabic = {
  group: (
    <g
      fillRule="evenodd"
      stroke="#000"
      strokeWidth=".335"
      style={{ stroke: '#fff' }}
    >
      <path d="M76.73.28v21.85H53.51V.28zm-4.62 14.58l-.37.23a10.85 10.85 0 0 1-5.53 1.69c-3.8 0-6.31-2.27-6.32-5.55s2.62-5.58 6.22-5.59a10.93 10.93 0 0 1 5.47 1.56l.36.2V4.47l-.16-.06a15.06 15.06 0 0 0-5.65-1.27 9.62 9.62 0 0 0-6.53 2.34 7.9 7.9 0 0 0-2.59 6 7.7 7.7 0 0 0 2.16 5.17A9.1 9.1 0 0 0 66 19.28a12.73 12.73 0 0 0 6-1.37l.14-.07zM50 .28v21.85H26.75V.28zm-5.54 14.48A4.21 4.21 0 0 0 41 10.69a3.87 3.87 0 0 0 1.35-1.08A3.41 3.41 0 0 0 43 7.47a3.8 3.8 0 0 0-1.27-2.85 5.76 5.76 0 0 0-4-1.29h-4.89v15.75h5.81a6.54 6.54 0 0 0 4.46-1.39 3.88 3.88 0 0 0 1.35-2.93z" />
      <path d="M41.57 14.45a1.86 1.86 0 0 1-.63 1.45 3.85 3.85 0 0 1-2.6.72h-2.68v-4.3h2.56a4.45 4.45 0 0 1 2.57.62 1.76 1.76 0 0 1 .78 1.51zM39.4 9.31a1.84 1.84 0 0 0 .74-1.63 1.66 1.66 0 0 0-.53-1.31 3 3 0 0 0-2.05-.57h-1.9v4.07H37a4.21 4.21 0 0 0 2.4-.56zM23.22.28v21.85H0V.28zM17.7 14.76a4.21 4.21 0 0 0-3.43-4.07 3.86 3.86 0 0 0 1.35-1.08 3.41 3.41 0 0 0 .65-2.13A3.8 3.8 0 0 0 15 4.63a5.76 5.76 0 0 0-4-1.29H6v15.74h5.9a6.54 6.54 0 0 0 4.46-1.39 3.89 3.89 0 0 0 1.34-2.93z" />
      <path d="M12.64 9.31a1.85 1.85 0 0 0 .74-1.63 1.66 1.66 0 0 0-.53-1.31 3 3 0 0 0-2.05-.57H8.91v4.07h1.3a4.21 4.21 0 0 0 2.43-.56zM14 12.94a4.45 4.45 0 0 0-2.57-.62H8.91v4.29h2.68a3.85 3.85 0 0 0 2.6-.72 1.86 1.86 0 0 0 .63-1.45 1.76 1.76 0 0 0-.82-1.5zM101 3.34h2.15v15.78h-1.94L90.69 7v12.12h-2.12V3.34h1.83L101 15.6zM107.26 3.34h8.95v2h-6.69v4.81H116v2h-6.46v4.9h6.9v2h-9.16zM140.33 3.34h2.25l-6.38 15.85h-.5l-5.16-12.84-5.21 12.84h-.49l-6.36-15.85h2.28l4.35 10.88 4.38-10.88h2.14L136 14.22zM148.7 12.51l-1.72-1a7.8 7.8 0 0 1-2.3-1.94 3.68 3.68 0 0 1-.68-2.2 3.88 3.88 0 0 1 1.29-3 4.83 4.83 0 0 1 3.36-1.16 6.36 6.36 0 0 1 3.63 1.11v2.49a5.23 5.23 0 0 0-3.67-1.64 3 3 0 0 0-1.82.51 1.55 1.55 0 0 0-.71 1.32 2 2 0 0 0 .52 1.33 6.6 6.6 0 0 0 1.69 1.3l1.73 1q2.89 1.73 2.89 4.39a4 4 0 0 1-1.27 3.08 4.65 4.65 0 0 1-3.3 1.19 6.94 6.94 0 0 1-4.26-1.44V15a5.32 5.32 0 0 0 4.24 2.32 2.66 2.66 0 0 0 1.77-.59 1.85 1.85 0 0 0 .71-1.48q-.02-1.45-2.1-2.74z" />
      <path d="M168.37 0v22.59" strokeWidth="2" />
      <path d="M203 4.64c-.11-.21-.26-.49-.32-.58h-.06L201 5.56v.06l.13.19.17.26a.79.79 0 0 0 .1.14 6.14 6.14 0 0 1 1.08 3.66v.25h-10.4l.42 1.76c0 .13 0 .2.3.2h2.86l-.16.35a4.36 4.36 0 0 1-4.26 2.7c-2.49 0-4.52-2-5.44-5.3l-.8 1.77a8.5 8.5 0 0 0 1.07 2.64 6.42 6.42 0 0 0 5.6 2.9 6.53 6.53 0 0 0 6.06-4.82v-.19h6.43c.11-1.07.16-1.53.16-1.92a6.12 6.12 0 0 0 0-.62A10.53 10.53 0 0 0 203 4.64M200.35 15.23a.27.27 0 0 0-.06.16.22.22 0 0 0 .05.13l1 1a.28.28 0 0 0 .16.06s.07 0 .16-.08l1-1a.24.24 0 0 0 .08-.13.21.21 0 0 0-.08-.16l-1-1a.18.18 0 0 0-.13-.06.25.25 0 0 0-.16.06zM190 18.36a.24.24 0 0 0-.16.06l-1 1a.18.18 0 0 0-.06.14.3.3 0 0 0 .08.19l1 1s.06.06.16.06a.18.18 0 0 0 .13-.06l1-1a.23.23 0 0 0 .05-.13.31.31 0 0 0-.06-.16l-1-1a.29.29 0 0 0-.15-.07" />
      <path d="M192.4 18.36a.3.3 0 0 0-.19.08l-1 1a.14.14 0 0 0-.06.14.3.3 0 0 0 .08.19l1 1a.27.27 0 0 0 .16.06h.11l1-1a.23.23 0 0 0 .08-.13.25.25 0 0 0-.08-.16l-1-1a.18.18 0 0 0-.13-.06M218.85 10.2a2.28 2.28 0 0 1-2.5-2.37c0-1.28.68-2.34 3.92-2.34h.55V4.31a13.17 13.17 0 0 0-2.49-.24c-2.43 0-4.2 1.45-4.2 3.45a3.72 3.72 0 0 0 .71 2.29l.26.39h-4.52v-.22a16.31 16.31 0 0 0-.84-3.62c-.14-.4-.34-1-.45-1.2l-1.7 1.55a.24.24 0 0 0 0 .06.71.71 0 0 0 .12.32 8.9 8.9 0 0 1 1.14 4.41 4.44 4.44 0 0 1-.55 2.37 4.27 4.27 0 0 1-3.3 1.83l1.49 1.42c2.52-.4 4-2.24 4.2-5.05a.78.78 0 0 0 .37.08h10.64c.21 0 .32-.06.32-.16V10.2z" />
    </g>
  ),
  ratio: width / height,
  viewbox: {
    height,
    width,
  },
};

export default arabic;
