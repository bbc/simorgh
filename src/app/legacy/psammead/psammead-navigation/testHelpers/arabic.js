const arabicNavData = [
  {
    title: 'رئيسية',
    url: '/arabic',
  },
  {
    title: 'شرق أوسط',
    url: '/arabic/middleeast',
  },
  {
    title: 'عالم',
    url: '/arabic/world',
  },
  {
    title: 'علوم وتكنولوجيا',
    url: '/arabic/scienceandtech',
  },
  {
    title: 'صحة',
    url: '/arabic/topics/c4794229-7f87-43ce-ac0a-6cfcd6d3cef2',
  },
  {
    title: 'اقتصاد',
    url: '/arabic/business',
  },
  {
    title: 'فنون',
    url: '/arabic/artandculture',
  },
  {
    title: 'رياضة',
    url: '/arabic/sports',
  },
  {
    title: 'مجلة',
    url: '/arabic/magazine',
  },
  {
    title: 'مرأة',
    url: '/arabic/topics/e45cb5f8-3c87-4ebd-ac1c-058e9be22862',
  },
  {
    title: 'فيديو',
    url: '/arabic/media',
  },
  {
    title: 'صحافة',
    url: '/arabic/inthepress',
  },
  {
    title: 'صور',
    url: '/arabic/media/photogalleries',
  },
  {
    title: 'برامجنا',
    url: '/arabic/tv-and-radio-37728883',
  },
  {
    title: 'ترند',
    url: '/arabic/trending',
  },
  {
    title: 'حوارات',
    url: '/arabic/interactivity',
  },
  {
    title: 'ملفات',
    url: '/arabic/in_depth',
  },
];

export default arabicNavData;
