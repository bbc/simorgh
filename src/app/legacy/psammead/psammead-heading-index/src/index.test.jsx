import React from 'react';
import { render } from '../../../../components/react-testing-library-with-providers';
import latin from '../../../../components/ThemeProvider/fontScripts/latin';
import arabic from '../../../../components/ThemeProvider/fontScripts/arabic';
import HeadingIndex from './index';

describe('Index Heading', () => {
  it('should render correctly', () => {
    const { container } = render(
      <HeadingIndex script={latin} service="news">
        This is a page heading
      </HeadingIndex>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with arabic script typography values', () => {
    const { container } = render(
      <HeadingIndex script={arabic} service="persian">
        هذا عنوان الصفحة
      </HeadingIndex>,
    );
    expect(container).toMatchSnapshot();
  });
});
