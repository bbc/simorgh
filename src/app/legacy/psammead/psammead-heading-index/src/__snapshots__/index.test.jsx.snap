// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Heading should render correctly 1`] = `
.emotion-0 {
  font-size: 1.125rem;
  line-height: 1.375rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #6E6E73;
  margin: 0;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-0 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-0 {
    font-size: 1.5rem;
    line-height: 1.75rem;
  }
}

<div>
  <h1
    class="emotion-0 emotion-1"
    tabindex="-1"
  >
    This is a page heading
  </h1>
</div>
`;

exports[`Index Heading should render correctly with arabic script typography values 1`] = `
.emotion-0 {
  font-size: 1.25rem;
  line-height: 1.75rem;
  font-family: "BBC Reith Qalam",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #6E6E73;
  margin: 0;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-0 {
    font-size: 1.25rem;
    line-height: 1.75rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-0 {
    font-size: 1.5rem;
    line-height: 2rem;
  }
}

<div>
  <h1
    class="emotion-0 emotion-1"
    tabindex="-1"
  >
    هذا عنوان الصفحة
  </h1>
</div>
`;
